### PR TITLE
Alerting: Support for alerting for react panels, and lots of fixes to alert annotations for both react and angular

### PR DESCRIPTION
**Pull request by torkelo**
* [x] Adds support for alerting for timeseries panel and make it work
* [x] Fixes alert annotations issues with angular graph
* [x] Fixes annotations rendering issues with time series, now shows correct text & state
* [x] Fixes annotation and alert annotation colors for GraphNG/Timeseries
* [x] Simplify PanelChromeAngular so that alert state is taken from PanelData instead of RenderEvent
* [x] Show alert state in React panel header

### CODE REVIEW

File: public/app/features/query/state/PanelQueryRunner.ts [hunk 1]
```diff
@@ -67,12 +66,10 @@ export class PanelQueryRunner {
   private subscription?: Unsubscribable;
   private lastResult?: PanelData;
   private dataConfigSource: DataConfigSource;
-  private dataSupport?: PanelPluginDataSupport;
 
   constructor(dataConfigSource: DataConfigSource) {
     this.subject = new ReplaySubject(1);
     this.dataConfigSource = dataConfigSource;
-    this.dataSupport = this.dataConfigSource.dataSupport;
```
Comment (torkelo) [inline public/app/features/query/state/PanelQueryRunner.ts:75]: I removed this as a property on dataConfigSource (PanelModel) as it was sometimes undefined when created before the plugin was loaded.

File: packages/grafana-data/src/types/data.ts [hunk 2]
```diff
@@ -169,7 +169,7 @@ export enum NullValueMode {
  */
 export interface DataConfigSource {
   configRev?: number;
-  dataSupport?: PanelPluginDataSupport;
+  getDataSupport: () => PanelPluginDataSupport;
```
Comment (torkelo) [inline packages/grafana-data/src/types/data.ts:172]: Removed this as a property, now PanelModel can just implement this function and return plugin.dataSupport , this change was due to dataSupport being undefined when going into panel edit, and in some or scenarios as well (I think)

**Review by hugohaggmark — APPROVED** [review]
Fantastic work and works as expected 🎉

File: public/app/features/dashboard/dashgrid/PanelChrome.tsx [hunk 3]
```diff
@@ -223,7 +223,7 @@ export class PanelChrome extends Component<Props, State> {
       panel.getQueryRunner().run({
         datasource: panel.datasource,
         queries: panel.targets,
-        panelId: panel.id,
+        panelId: panel.editSourceId || panel.id,
```
Comment (torkelo) [inline public/app/features/dashboard/dashgrid/PanelChrome.tsx:226]: I also introduced this change to fix annotations showing up in panel edit.

**Pull request merged**